src/__tests__/pet.test.js

```diff
@@ -0,0 +1,168 @@
+const { Pet } = require ('../pet.js');
+
+const ERROR_MESSAGE = 'Your pet is no longer alive :(';
+
+let fido;
+beforeEach(() => {
+    fido = new Pet('Fido');
+});
+
+describe ('Pet', () => {
+
+  test('Returns an instance of Pet', () => {
+    expect(fido).toBeInstanceOf(Pet);
+  });
+
+  test('Returns the pets name', () => {
+    expect(fido.name).toBe('Fido');
+  });
+});
+
+describe ('growUp', () => {
+  
+  test ('has an initial age of 0', () => {
+    expect (fido.age).toEqual (0);
+   });
+
+   test ('increments the age by 1', () => {
+    fido.growUp();
+    expect (fido.age).toEqual (1);
+  });
+  
+  test ('has an initial hunger of 0', () => {
+    expect (fido.hunger).toEqual (0);
+  })
+
+  test ('increments the hunger by 5', () => {
+    fido.growUp();
+    expect (fido.hunger).toEqual (5);
+  });
+
+  test ('has an initial fitness of 10', () => {
+    expect (fido.fitness).toEqual (10);
+  });
+
+  test ('decrements the fitness by 3', () => {
+    fido.growUp();
+    expect (fido.fitness).toEqual (7);
+  });
+
+  test(`growUp should throw error: "${ERROR_MESSAGE}"`, () => {
```

Comment:
This test name is a bit misleading. It currently reads as though the growUp method should always throw an error, so we have to be more specific in what we are doing here. We expect it to throw an error when certain conditions are met. We should include this in the test title.